src/ast/ast_build_filter_tree.c

```diff
@@ -159,15 +200,31 @@ static FT_FilterNode *_convertInlinedProperties(RecordMap *record_map, const AST
 FT_FilterNode *_FilterNode_FromAST(RecordMap *record_map, const cypher_astnode_t *expr) {
 	assert(expr);
 	cypher_astnode_type_t type = cypher_astnode_type(expr);
-	if(type == CYPHER_AST_BINARY_OPERATOR) {
-		return _convertBinaryOperator(record_map, expr);
+
+	if(type == CYPHER_AST_NODE_PATTERN) {
+		assert(false);
+		// return _convertInlinedProperties(record_map, ast, expr, expr_NODE);
+	} else if(type == CYPHER_AST_REL_PATTERN) {
+		assert(false);
+		// return _convertInlinedProperties(record_map, ast, expr, expr_EDGE);
 	} else if(type == CYPHER_AST_COMPARISON) {
 		return _convertComparison(record_map, expr);
+	} else if(type == CYPHER_AST_BINARY_OPERATOR) {
+		return _convertBinaryOperator(record_map, expr);
+	} else if(type == CYPHER_AST_UNARY_OPERATOR) {
+		return _convertUnaryOperator(record_map, expr);
 	} else if(type == CYPHER_AST_APPLY_OPERATOR) {
-		assert(false && "Unary APPLY operators are not currently supported in filters.");
+		return _convertUnaryOperator(record_map, expr);
```

Comment:
Does this correctly handle some cases? It is definitely not right for queries like:
`"MATCH (a) WHERE NOT EXISTS(a.name) RETURN a"`
`Assertion `cypher_astnode_instanceof(astnode, CYPHER_AST_UNARY_OPERATOR)' failed.'

Reply:
Great catch! fixed.